rest/src/protocols.cpp

```diff
@@ -22,27 +22,28 @@
 #include "protocols.h"
 #include "commands.h"
 #include "message-bus.h"
+#include <fty/rest/component.h>
 #include <tnt/http.h>
 
 namespace fty {
 
 unsigned Protocols::run()
 {
-    auto user = global<UserInfo>("UserInfo user");
-    if (auto ret = checkPermissions(**user, m_permissions); !ret) {
+    rest::User user(m_request);
+    if (auto ret = checkPermissions(user.profile(), m_permissions); !ret) {
         throw rest::Error(ret.error());
     }
 
     commands::protocols::In param;
-    if (auto res = pack::json::deserialize(m_request.getBody(), param); !res) {
-        throw rest::Error("bad-input", m_request.getArg(0));
+    if (auto res = pack::json::deserialize(m_request.body(), param); !res) {
+        throw rest::errors::BadRequestDocument(res.error());
     }
 
     if (auto list = protocols(param)) {
-        m_reply.out() << *list << "\n\n";
+        m_reply << *list << "\n\n";
         return HTTP_OK;
     } else {
-        throw rest::Error("internal-error", list.error());
+        throw rest::errors::Internal(list.error());
```

Comment:
I would like to have this error translated instead of having internal error

Reply:
This error would be translated :)